library/src/main/java/org/apache/apex/malhar/lib/fs/s3output/S3InitiateFileUpload.java

```diff
@@ -0,0 +1,378 @@
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.google.common.base.Preconditions;
+
+import com.datatorrent.api.Context;
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.api.Operator;
+import com.datatorrent.lib.io.fs.AbstractFileSplitter;
+
+/**
+ * This is an S3 Initiate file upload operator which can be used to initiate file upload and emits the upload id.
+ * Initiate the given file for upload only if the file contains more than one block.
+ *
+ */
+@InterfaceStability.Evolving
+public class S3InitiateFileUpload implements Operator, Operator.CheckpointNotificationListener
+{
+  @NotNull
+  private String bucketName;
+  @NotNull
+  private String accessKey;
+  @NotNull
+  private String secretAccessKey;
+  private String endPoint;
+  @NotNull
+  private String outputDirectoryPath;
+  private WindowDataManager windowDataManager = new FSWindowDataManager();
+  protected transient AmazonS3 s3Client;
+  protected transient long currentWindowId;
+  protected transient List<UploadFileMetadata> currentWindowRecoveryState;
+
+  public final transient DefaultOutputPort<UploadFileMetadata> fileMetadataOutput = new DefaultOutputPort<>();
+
+  public final transient DefaultOutputPort<UploadFileMetadata> uploadMetadataOutput = new DefaultOutputPort<>();
+
+  /**
+   * This input port receive file metadata and those files will be upload into S3.
+   */
+  public final transient DefaultInputPort<AbstractFileSplitter.FileMetadata> filesMetadataInput = new DefaultInputPort<AbstractFileSplitter.FileMetadata>()
+  {
+    @Override
+    public void process(AbstractFileSplitter.FileMetadata tuple)
+    {
+      processTuple(tuple);
+    }
+  };
+
+  /**
+   * For the input file, initiate the upload and emit the UploadFileMetadata through the fileMetadataOutput,
+   * uploadMetadataOutput ports.
+   * @param tuple given tuple
+   */
+  protected void processTuple(AbstractFileSplitter.FileMetadata tuple)
+  {
+    if (currentWindowId <= windowDataManager.getLargestCompletedWindow()) {
+      return;
+    }
+    String keyName = getKeyName(tuple.getFilePath());
+    String uploadId = "";
+    if (tuple.getNumberOfBlocks() > 1) {
+      InitiateMultipartUploadRequest initRequest = new InitiateMultipartUploadRequest(bucketName, keyName);
+      initRequest.setObjectMetadata(createObjectMetadata());
+      InitiateMultipartUploadResult initResponse = s3Client.initiateMultipartUpload(initRequest);
+      uploadId = initResponse.getUploadId();
+    }
+    UploadFileMetadata uploadFileMetadata = new UploadFileMetadata(tuple, uploadId, keyName);
+    fileMetadataOutput.emit(uploadFileMetadata);
+    uploadMetadataOutput.emit(uploadFileMetadata);
+    currentWindowRecoveryState.add(uploadFileMetadata);
+  }
+
+  /**
+   * Creates the empty object metadata for initiate multipart upload request.
+   * @return the ObjectMetadata
+   */
+  public ObjectMetadata createObjectMetadata()
```

Comment:
No. By default it creates empty objectMetadata. User can override this and set the properties like encryption algorithm, etc. For more details, please refer below link:
http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/model/ObjectMetadata.html